pkg/core/proxy/integrations/mongo/scramAuth.go

```diff
@@ -368,8 +373,8 @@ func handleSaslContinue(actualMsg map[string]interface{}, responseSection, mongo
 	fields := strings.Split(string(decodedResponsePayload), ",")
 	verifier, err := parseFieldBase64(fields[0], "v")
 	if err != nil {
-		utils.LogError(logger, err, "failed to parse the verifier of final response message")
-		return "", false, err
+		logger.Debug("failed to parse the verifier of final response message")
```

Comment:
incase of  mongo 3.x after the sasl continue is done, client sends some more requests that have data like ~

```
map[$db:admin conversationId:map[$numberInt:1] payload:map[$binary:map[base64: subType:00]]
```

Since the mongo message has an saslcontinue in the body handlesaslcontinue is triggered, but as this doesn't has any scram related data (server proof in this case, the old code would return error breaking the flow)

Reply:
do proper error handling.